app/uk/gov/hmrc/helptosave/util/Logging.scala

```diff
@@ -29,42 +32,52 @@ object Logging {
 
   implicit class LoggerOps(val logger: Logger) {
 
-    def debug(message: String, nino: NINO)(implicit transformer: NINOLogMessageTransformer): Unit =
-      logger.debug(transformer.transform(message, nino))
+    def debug(message: String, nino: NINO)(implicit transformer: LogMessageTransformer, hc: HeaderCarrier): Unit =
```

Comment:
where we're using a `Logger` we may not always have an instance of `HeaderCarrier` - making a `HeaderCarrier` an implicit parameter here restricts the use of the methods in `LoggerOps`. I think it'd be more flexible if we created extra methods in `LoggerOps` to pass in `correlationId: Option[UUID]` with perhaps an default value of `None` if we don't want to create extra noise

Reply:
The same goes for the proxy PR as well